### PR TITLE
feat: assignment eight memory management

### PR DESCRIPTION
**Pull request by Sruthi-Subaraja**
ASSIGNMENT 8 - MEMORY MANAGEMENT
TASK 1 - Understanding and using value types and reference types in C#.
TASK 2 - Working with the stack and the heap.
TASK 3 - Using garbage collection and understanding it's impact on performance.
TASK 4 - Implementing and understanding the IDisposable interface and the 'using' statement

### CODE REVIEW

**Review by Tharun-Sundar — CHANGES_REQUESTED** [review]

File: src/IDisposableDemo/FileClass.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿// <copyright file="FileClass.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace IDisposableDemo
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileClass.cs:5]: There is a separate assignment for IDisposable design pattern, For this assignment, implementation of dispose method only is required for this assignment.

File: src/IDisposableDemo/FileClass.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /// <summary>
+    /// File class to read and write to a text file
+    /// </summary>
+    public class FileClass : IDisposable
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileClass.cs:10]: Done

File: src/IDisposableDemo/FileClass.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        /// <summary>
+        /// Gets or sets stream writer object
+        /// </summary>
+        /// <value>Gstream writer object</value>
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileClass.cs:15]: It's a typo. I've changed it.

File: src/IDisposableDemo/FileClass.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        public StreamWriter? Sw { get; set; }
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileClass.cs:16]: Done.

File: src/IDisposableDemo/Program.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            using (StreamWriter sw = new StreamWriter("D:\\Csharp\\src\\IDisposableDemo\\textFile.txt"))
+            {
+                FileClass fileClass = new FileClass();
+                Console.WriteLine("Write any text");
+                string? myStory = Console.ReadLine();
+                sw.WriteLine(myStory);
+            }
+
+            StreamReader sr = new StreamReader("D:\\Csharp\\src\\IDisposableDemo\\textFile.txt");
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/Program.cs:26]: If I use relative path, then the path won't be constant.

File: src/IDisposableDemo/Program.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    /// </summary>
+    public class Program
+    {
+        /// <summary>
+        /// Main method
+        /// </summary>
+        /// <param name="args">String array of arguments</param>
+        public static void Main(string[] args)
+        {
+            using (StreamWriter sw = new StreamWriter("D:\\Csharp\\src\\IDisposableDemo\\textFile.txt"))
```
[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/Program.cs:18]: Okay

[the remark this answers is not in the document]
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/Program.cs:18]: Yes, even when writing to a file in a separate method, Using is needed to close the file.

File: src/IDisposableDemo/FileHandler.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        public FileHandler(string filePath)
+        {
+            this.FileWriter = new StreamWriter(filePath);
+        }
+
+        /// <summary>
+        /// Gets or sets stream writer object.
+        /// </summary>
+        /// <value>Stream writer object</value>
+        public StreamWriter FileWriter { get; set; }
```
Comment (Tharun-Sundar) [inline src/IDisposableDemo/FileHandler.cs:24]: Remove this property

File: src/IDisposableDemo/FileClass.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+﻿// <copyright file="FileClass.cs" company="PlaceholderCompany">
```
Comment (Tharun-Sundar) [inline src/IDisposableDemo/FileClass.cs:1]: Remove this class

**Review by Tharun-Sundar — APPROVED** [review]

File: src/IDisposableDemo/FileClass.cs (hunk 8 above)
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileClass.cs:1]: Okay

File: src/IDisposableDemo/FileHandler.cs (hunk 7 above)
Comment (Sruthi-Subaraja) [reply src/IDisposableDemo/FileHandler.cs:24]: Okay

**Pull request merged**